### ISSUE TITLE
Optimization: Implement constant pools on MIPS64R6 using LWPC and LDPC instruction

### ISSUE THREAD

**Issue by ibogosavljevic** [v8mips/v8mips#401]

**Comment by sretenkovacevic** [comment]
Idea was to implement Constant Pool to work with 64bit constants, using LDPC instruction. It was expected that the most benefits of Constant Pool would be seen if it was used on 64bit constants because load of those constants needed the most instructions without it.

Implementation of Constant Pool was ported from ARM64 with some minor modifications and adjustments.
Optimization was implemented on commit with ID 6c9ef89540f027a625d9d7f87f483f868ba45283.
Implementation itself is very complex, changes were made in 19 different files and demanded work on
various parts of code generation and optimization.

With Constant Pool generated code is smaller and less instructions are executed.
benchmarks/octane tests were run with and without Constant Pool five times each as a performance test. Average results show that with Constant Pool there is performance drop for 6.46%. Not in a single situation implementation with Constant Pool had better results. Therefore we decided that it's not justified to commit this patch, so this issue is closed. 

Test records and patch are attached.



[Constant Pool.zip](https://github.com/v8mips/v8mips/files/678557/Constant.Pool.zip)
